### PR TITLE
Pedsim Observation Unit

### PR DESCRIPTION
**Pull request by tuananhroman**
- Implemented a new observation collector unit for collecting and processing pedsim data.
- Updated the DRL Agent Node to use the new ObservationManager
- Renamed/removed redundant files
- Extensive code documentation

### CODE REVIEW

File: testing/scripts/drl_agent_node.py [hunk 1]
```diff
@@ -22,27 +22,23 @@
...
         """
-        ns = remove_double_slash(ns)
-        self.observation_collector = ObservationCollector(
-            ns, rospy.get_param("laser/num_beams"), external_time_sync=False
-        )
+        self.ns = ns.remove_double_slash()
+        self.observation_collector = ObservationManager(self.ns)
 
-        self._max_laser_range = rospy.get_param(os.path.join(ns, "laser", "range"))
+        self._max_laser_range = self.ns("laser", "range")
```
Comment (voshch) [inline testing/scripts/drl_agent_node.py:35]: shouldn't this be `rospy.get_param(self.ns("laser", "range"))`?

File: utils/misc/rl_utils/rl_utils/utils/observation_collector/observation_units/pedsim_collector_unit.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        """
+        Static method to extract agent positions from a list of agent states.
+
+        Args:
+            agent_states (List[AgentState]): A list of agent states.
+
+        Returns:
+            List[Pose]: A list of agent poses.
+        """
+        return [agent.pose for agent in agent_states]
```
Comment (voshch) [inline utils/misc/rl_utils/rl_utils/utils/observation_collector/observation_units/pedsim_collector_unit.py:112]: Keep in mind that `agent.pose.orientation` is (currently) bogus data. The correct yaw can be calculated from the velocity vector (which is also consistent with Gazebo) when needed.

**Review by voshch — COMMENTED** [review]
one question, one note

**Comment by tuananhroman** [comment]
@linhdoan8

**Review by linhdoan8 — APPROVED** [review]
all ok

**Pull request merged**